src/custom-markdown-features/critic-markup.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * critic markup	              HTML         	          LaTeX
+ * {--[text]--}	          <del>[text]</del>	                    \st{[text]}
+ * {++[text]++}          	<ins>[text]</ins>	                    \underline{[text]}
+ * {~~[text1]~>[text2]~~}	<del>[text1]</del><ins>[text2]</ins>	\st{[text1]}\underline{[text2]}
+ * {==[text]==}          	<mark>[text]</mark>	                  \hl{[text]}
+ * {>>[text]<<}          	<aside>[text]</aside>                	\marginpar{[text]}
+*/
+import { MarkdownIt } from 'markdown-it'
+import { MarkdownEngineConfig } from '../markdown-engine-config'
+import parseMath from '../parse-math'
+
+export default (md: MarkdownIt, config: MarkdownEngineConfig) => {
+  // @ts-ignore
+  md.inline.ruler.before('strikethrough', 'critic-markup', (state, silent) => {
```

Comment:
@shd101wyy could you please explain why this function returns boolean? According to a [type definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f1f971c41a7a98820b8439ef206b98a380982079/types/markdown-it/index.d.ts#L99-L101) in [@types/markdown-it](https://www.npmjs.com/package/@types/markdown-it) it is expected to return void. I had to add `// @ts-ignore` in several feature files to ignore build errors meanwhile. Is there any place in mume or markdown-it that is using the returned true / false? I could not find it.